packages/opentelemetry-core/src/common/ConsoleLogger.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger } from '@opentelemetry/types';
+import * as consoleLogLevel from 'console-log-level';
+import * as util from 'util';
+
+/** Defines a logger options interface. */
+export interface LoggerOptions {
+  /** level of logger - 0:disable, 1: error, 2: warn, 3: info, 4: debug  */
+  level?: string;
+}
+export type ConsoleLogLevel = 'error' | 'warn' | 'info' | 'debug';
+export type LogLevel = 'silent' | ConsoleLogLevel;
+
+/** The list of log levels. */
+export const LEVELS: ReadonlyArray<LogLevel> = [
+  'silent',
+  'error',
+  'warn',
+  'info',
+  'debug',
+];
+
+export class ConsoleLogger implements Logger {
+  private _logger: consoleLogLevel.Logger;
+  level?: string;
+  constructor(options?: LoggerOptions | string | number) {
+    this.level = this._logLevelToName(options).level || 'debug';
+    this._logger = consoleLogLevel({
+      stderr: true,
+      level: this.level as ConsoleLogLevel,
+    });
+  }
+
+  debug(message: string, ...args: unknown[]) {
+    this._logger.debug(util.format(message, ...args));
```

Comment:
Calling `sprintf` can be expensive as it happens before log level decision. So you will pay the cost for every call no matter if it will log or not. We burn our self earlier with it: https://github.com/joyent/node-verror/issues/63

Reply:
It's also not supported by many loggers, I think it would be an acceptable tradeoff to allow passing only strings and objects to the logger and log them as it is without pre processing. like: 

- non-structured log: `logger.info('zipkin exporter send result is "success"')`
- structured log: `logger.info({ exporter: "zipkin", event: "send",  result: "success",  transport: "http",  protocol: "http",  method: "get",  uri: "http://..."   retry_count: 1,   retry_limit: 3, ... }`)`

Reply:
What if we did something where we change the functions based on the log level, e.g.:

```
const NO_OP: LogFunction = (...args) => {};

class ConsoleLogger {
   readonly debug: LogFunction;
   readonly info: LogFunction;
   ...

  constructor(level: LogLevel) {
    if (level >= LogLevel.DEBUG) {
      this.debug = (...args) => { console.debug(args) };
    } else {
      this.debug = NO_OP;
    }
    if (level >= LogLevel.INFO) {
       this.info = (...args) => { console.info(args) };
    } else {
       this.info = NO_OP;
    }
  ...
}
```

That way there is minimal penalty at log time since we are just directly delegating to `console` or doing a no-op.

Reply:
I have made the changes as per your suggestions, please take a look. Thanks :)